### PR TITLE
Job observable

### PR DESCRIPTION
**Pull request by bildja**

### CODE REVIEW

File: src/platform/authentication/token.ts [hunk 1]
```diff
@@ -20,7 +20,7 @@ export class Token {
    * @description a policy like objects array
    * @type {Array<Array<{}>>}
    */
-  public objects: TokenObjects | null = null;
+  public objects: TokenObjects | null = null || null;
```
Comment (ArtemAstakhovWix) [inline src/platform/authentication/token.ts]: null || null?

File: src/platform/management/files-search.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    this.apiUrl = `${this.baseUrl}/_api/files/search`;
+  }
+
+  /**
+   * @param {function(null, IImageExtractionResponse)} callback
+   */
+  private parseResponse(callback?: (error: null, params: IImageExtractionResponse) => void): (response: RawResponse<ImageExtractionResponse>) => ImageExtractionResponse {
+    return (response: RawResponse<ImageExtractionResponse>) => {
+      return response.payload;
+      // const imageExtractionResponse = new ImageExtractionResponse(response.payload);
```
Comment (ArtemAstakhovWix) [inline src/platform/management/files-search.ts]: remove?

File: src/platform/management/job-manager.ts [hunk 3]
```diff
@@ -54,12 +54,12 @@ export class JobManager {
    * @param {string} groupId
    * @param {function(Error, Job[])} callback DEPRECATED! use promise response instead
    */
-  getJobGroup(groupId, callback?: (error: Error | null, job: Job<any>[] | null) => void): Promise<Job<any>[]> {
+  getJobGroup<T = any>(groupId, callback?: (error: Error | null, job: Job<any>[] | null) => void): Promise<Job<any>[]> {
```
Comment (ArtemAstakhovWix) [inline src/platform/management/job-manager.ts]: any?

Comment (bildja) [reply src/platform/management/job-manager.ts]: @ArtemAstakhovWix yes, I had an explanation in a different file, will copy it here, I can not fix it now, the typings need to be reformatted a bit

File: src/platform/authentication/token.ts (hunk 1 above)
Comment (bildja) [reply src/platform/authentication/token.ts]: No idea how it happens, I haven't even edited this file :)

**Pull request merged**